### PR TITLE
Add CosmosDb endpoint and fix ACR endpoints for Gov Clouds

### LINKED ISSUE #366: Expose 'CosmosDBDNSSuffix' in the Environments struct
👋 

We're working through scoping out what's needed to add support for CosmosDB Collections & Databases into Terraform. As a part of the investigation here we've noticed that there's no suffix available as part of the Environment struct for the CosmosDB DNS Suffix, [as there is for example for Container Registries etc](https://github.com/Azure/go-autorest/blob/master/autorest/azure/environments.go#L37-L57).

Would it be possible to add a new property to the Environments struct which exposes the CosmosDB DNS Suffix?

Thanks!

### PR DESCRIPTION
**Pull request by vladbarosan**
- fix DNS endpoint for ACR
- Add Cosmos DB endpoint

Fixes #366

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Can you merge https://github.com/Azure/go-autorest/pull/376 into this PR?

File: autorest/azure/environments.go [hunk 1]
```diff
@@ -150,7 +154,8 @@ var (
 		ServiceBusEndpointSuffix:     "servicebus.cloudapi.de",
 		ServiceManagementVMDNSSuffix: "azurecloudapp.de",
 		ResourceManagerVMDNSSuffix:   "cloudapp.microsoftazure.de",
-		ContainerRegistryDNSSuffix:   "azurecr.io",
+		ContainerRegistryDNSSuffix:   "",
```
Comment (jhendrixMSFT) [inline autorest/azure/environments.go]: This is supposed to be empty?

**Comment by vladbarosan** [comment]
@jhendrixMSFT  it seems ACR is not present in the German Cloud . do we want to leave it blank ?

File: autorest/azure/environments.go (hunk 1 above)
Comment (vladbarosan) [reply autorest/azure/environments.go]: beat me to it, there is no endpoint in [German Cloud]( https://azure.microsoft.com/en-us/global-infrastructure/services/?regions=china-non-regional,china-east,china-east-2,china-north,china-north-2,germany-non-regional,germany-central,germany-northeast,us-central,us-east,us-east-2,us-north-central,us-south-central,us-west-central,us-west,us-west-2&products=all )  ( also searched internally )

Comment (jhendrixMSFT) [reply autorest/azure/environments.go]: So if we omit it entirely and somebody tried to use this for German configuration will they get a compile time error?  I would prefer that to possible weird behavior due to empty string.  Or should the value be the one for public?

Comment (vladbarosan) [reply autorest/azure/environments.go]: removed per our discussion

**Comment by vladbarosan** [comment]
Fixes #375

**Pull request merged**